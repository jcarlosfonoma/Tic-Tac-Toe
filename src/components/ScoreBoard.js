import { useContext } from "react";
import MyContext from "../store/matches-record-context";
import styled from "styled-components";

const ScoreContainer = styled.div`
  display: flex;
  justify-content: space-between;
  color: rgb(61, 66, 80);
  width: 326px;
  margin-right: 4px;
  margin-bottom: 10px;
`;

const Score = styled.span`
  font-weight: bold;
  font-size: 20px;
`;

const ScoreBoard = () => {
  const matchesResultCtx = useContext(MyContext);
  return (
    <ScoreContainer>
      <Score data-testid="x-score-span">{`X Wins: ${matchesResultCtx.xWins}`}</Score>
      <Score data-testid="o-score-span">{`O Wins: ${matchesResultCtx.oWins}`}</Score>
      <Score data-testid="draws-span">{`Draws: ${matchesResultCtx.draws}`}</Score>
    </ScoreContainer>
  );
};

export default ScoreBoard;
