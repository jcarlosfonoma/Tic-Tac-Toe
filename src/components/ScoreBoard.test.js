import { cleanup, render } from "@testing-library/react";
import ScoreBoard from "./ScoreBoard";

describe("Square component", () => {
  afterEach(cleanup);

  it("Square should take a snapshot", () => {
    const { asFragment } = render(<ScoreBoard />);
    expect(asFragment(<ScoreBoard />)).toMatchSnapshot();
  });

  it("Checks if initial X score is equal to X Wins: 0", () => {
    const { getByTestId } = render(<ScoreBoard />);
    expect(getByTestId("x-score-span")).toHaveTextContent("X Wins: 0");
  });

  it("Checks if initial O score is equal to O Wins: 0", () => {
    const { getByTestId } = render(<ScoreBoard />);
    expect(getByTestId("o-score-span")).toHaveTextContent("O Wins: 0");
  });

  it("Checks if initial Draws score is equal to Draws: 0", () => {
    const { getByTestId } = render(<ScoreBoard />);
    expect(getByTestId("draws-span")).toHaveTextContent("Draws: 0");
  });
});
