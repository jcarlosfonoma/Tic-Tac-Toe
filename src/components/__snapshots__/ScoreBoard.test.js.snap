// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square component Square should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL cYbEbw"
  >
    <span
      class="sc-gsDKAQ dTgrTI"
      data-testid="x-score-span"
    >
      X Wins: 0
    </span>
    <span
      class="sc-gsDKAQ dTgrTI"
      data-testid="o-score-span"
    >
      O Wins: 0
    </span>
    <span
      class="sc-gsDKAQ dTgrTI"
      data-testid="draws-span"
    >
      Draws: 0
    </span>
  </div>
</DocumentFragment>
`;
